forage-android/src/main/java/com/joinforage/forage/android/ui/ForagePINEditText.kt

```diff
@@ -25,48 +25,71 @@ class ForagePINEditText @JvmOverloads constructor(
     attrs: AttributeSet? = null,
     defStyleAttr: Int = R.attr.foragePanEditTextStyle
 ) : AbstractForageElement(context, attrs, defStyleAttr) {
-    private var vault: VaultWrapper
+    private val _linearLayout: LinearLayout
+    private val btVaultWrapper: BTVaultWrapper
+    private val vgsVaultWrapper: VGSVaultWrapper
+    private var _SET_ONLY_vault: VaultWrapper? = null
+    private val vault: VaultWrapper
+        get() {
+            if (_SET_ONLY_vault == null) {
+                TODO("throw an error here since that means setForageContext was not called")
+                throw Exception("need to call setForageContext first!")
+            }
+            return _SET_ONLY_vault!!
+        }
 
     init {
         context.obtainStyledAttributes(attrs, R.styleable.ForagePINEditText, defStyleAttr, 0)
             .apply {
                 try {
-                    // Must initialize DD at the beginning of each render function. DD requires the context,
-                    // so we need to wait until a context is present to run initialization code. However,
-                    // we have logging all over the SDK that relies on the render happening first.
-                    val logger = Log.getInstance()
-                    logger.initializeDD(context)
                     setWillNotDraw(false)
                     orientation = VERTICAL
                     gravity = Gravity.CENTER
 
-                    val vaultType = LDManager.getVaultProvider(context.applicationContext as Application, logger)
-                    vault = if (vaultType == VaultConstants.BT_VAULT_TYPE) {
-                        BTVaultWrapper(context, attrs, defStyleAttr)
-                    } else {
-                        VGSVaultWrapper(context, attrs, defStyleAttr)
-                    }
+                    btVaultWrapper = BTVaultWrapper(context, attrs, defStyleAttr)
+                    vgsVaultWrapper = VGSVaultWrapper(context, attrs, defStyleAttr)
```

Comment:
probably should leave this as a commit (I definitely touch on it in the commit history) but we can initialize them both and can decide which one to add to the UI later!

Reply:
created a comment in [b188107](https://github.com/teamforage/forage-android-sdk/pull/103/commits/b188107e9d92953befbecc503bbbffb52cb0a814)